### DISCUSSION TITLE
Best way to deal with ARFLAGS

### DISCUSSION THREAD

**Discussion by jwbee** [bazelbuild/rules_foreign_cc#953] [Q&A]
Yesterday I needed to bazelize my build of icu4c and the way `configure_make` emits `ARFLAGS` to the environment, and the way ICU consumes `ARFLAGS`, are not compatible. Specifically, right here:

https://github.com/unicode-org/icu/blob/289d9703a03494ef0d9dfec122169b22bd9fc84f/icu4c/source/icudefs.mk.in#L120

```
ARFLAGS = @ARFLAGS@ r
```

With typical toolchain setup, this results in an invocation `ar rcsD r name.a name.o` which is not valid. I don't think I'm the first person to face this problem, as I found this hack for building rdkafka with `configure_make` at https://github.com/envoyproxy/envoy/blob/main/bazel/foreign_cc/BUILD#L65. I worked around my ICU issue with a similar hack: making `ARFLAGS=` be the first argument to `configure` so it overrides the `ARFLAGS` that bazel is setting in the environment. 

The whole issue seems a bit gross, though. Is the argument format of `ar` just impossible to deal with? Is this best fixed in ICU or in bazel?

**Comment by attilaolah** [comment]
A bit late to join here, but instead of specifying `ARFLAGS` as part of the `targets`, you could instead pass `args = ["--environment-overrides"]` to `make(…)`. That way, the already-defined `ARFLAGS` from the environment should take precedence over the ones coming from the Make files.

Not sure whether this would resolve your issue, but it did seem to work for me in another package.
